packages/devtools_app/lib/src/http/http_request_data.dart

```diff
@@ -192,6 +192,23 @@ class HttpRequestData extends NetworkRequest {
     return connectionInfo != null ? connectionInfo[_localPortKey] : null;
   }
 
+  @override
+  bool get didFail {
+    if (status == null) return false;
+    try {
+      final code = int.parse(status);
+      // Status codes 400-499 are client errors and 500-599 are server errors.
+      if (code >= 400) {
+        return true;
+      }
+    } on Exception catch (_) {
+      if (status == 'Error') {
```

Comment:
handle this before the int.parse. DevTools is good at avoiding throwing exceptions in normal operation cases and it would be nice to keep it that way.